augur/filter/__init__.py

```diff
@@ -63,11 +64,17 @@ def register_arguments(parser):
     Since priorities represent relative values between strains, these values can be arbitrary.""")
     subsample_group.add_argument('--subsample-seed', type=int, help="random number generator seed to allow reproducible subsampling (with same input data).")
 
-    output_group = parser.add_argument_group("outputs", "possible representations of filtered data (at least one required)")
+    output_group = parser.add_argument_group("outputs", "options related to outputs, at least one of the possible representations of filtered data (--output, --output-metadata, --output-strains) is required")
     output_group.add_argument('--output', '--output-sequences', '-o', help="filtered sequences in FASTA format")
     output_group.add_argument('--output-metadata', help="metadata for strains that passed filters")
     output_group.add_argument('--output-strains', help="list of strains that passed filters (no header)")
     output_group.add_argument('--output-log', help="tab-delimited file with one row for each filtered strain and the reason it was filtered. Keyword arguments used for a given filter are reported in JSON format in a `kwargs` column.")
+    output_group.add_argument(
+        '--empty-results-reporting',
+        type=EmptyResultsReportingMethod,
+        choices=list(EmptyResultsReportingMethod),
```

Comment:
The default argparse behavior when an argument has a list of `choices` and the user provides an invalid choice is to print an error that lists the valid choices. For example, if I run augur filter with `--exclude-ambiguous-dates-by fake`, I get the error:

```
augur filter: error: argument --exclude-ambiguous-dates-by: invalid choice: 'fake' (choose from 'any', 'day', 'month', 'year')
```

I wish that this implementation for results reporting behaved the same way, but instead if I run augur filter with `--empty-results-reporting fake`, I get:

```
augur filter: error: argument --empty-results-reporting: invalid EmptyResultsReportingMethod value: 'fake'
```

I hadn't thought about this specific type of implementation before now, so I checked [the Python argparse docs for `choices`](https://docs.python.org/3/library/argparse.html#choices) for examples and found:

> Use of [enum.Enum](https://docs.python.org/3/library/enum.html#enum.Enum) is not recommended because it is difficult to control its appearance in usage, help, and error messages.

What if we dropped the enum class and used a list of choice strings `["error", "warn", "skip"]` instead? Is there a strong argument to use the enum class instead?

Reply:
The choice of using enums was to make development and maintenance easier for us. 
It helps us standardize the choice values, we no longer have to check arg values with "magic" strings, and I think it makes the code more readable. 

I can update the `ArgparseEnum` class with a custom method to make this error message more useful.

Reply:
See custom type converter added in 4decbe46fa1c6d816ff2dca613b811791a2a2737. 

This now prints error as:
```
augur filter: error: argument --empty-results-reporting: invalid choice: 'fake' (choose from 'error', 'warn', 'skip')
```

Reply:
Got it! Thank you for the improved error message, @joverlee521. It looks like it wasn't quite as bad as the Python docs made it sound, after all!